cmd/resource-generator/main.go

```diff
@@ -40,6 +40,7 @@ type templateFields struct {
 	Year                   string
 	Name                   string
 	NameLower              string
+	IsNotNamed             bool
```

Comment:
I'm not a big fan of new field's name, but I can't come up with a better name. Perhaps we could just make `getResourceName()` gracefully handle the case where the spec does not have a name instead?